src/app/organizations/manage/reset-password.component.ts

```diff
@@ -0,0 +1,196 @@
+    }
+
+    getPasswordScoreAlertDisplay() {
+        if (this.enforcedPolicyOptions == null) {
+            return '';
+        }
+
+        let str: string;
+        switch (this.enforcedPolicyOptions.minComplexity) {
+            case 4:
+                str = this.i18nService.t('strong');
+                break;
+            case 3:
+                str = this.i18nService.t('good');
+                break;
+            default:
+                str = this.i18nService.t('weak');
+                break;
+        }
+        return str + ' (' + this.enforcedPolicyOptions.minComplexity + ')';
+    }
+
+    async generatePassword() {
+        const options = (await this.passwordGenerationService.getOptions())[0];
+        this.newPassword = await this.passwordGenerationService.generatePassword(options);
+        this.updatePasswordStrength();
+    }
+
+    togglePassword() {
+        this.showPassword = !this.showPassword;
+        document.getElementById('newPassword').focus();
+    }
+
+    copy(value: string) {
+        if (value == null) {
+            return;
+        }
+
+        this.platformUtilsService.copyToClipboard(value, { window: window });
+        this.platformUtilsService.showToast('info', null,
+            this.i18nService.t('valueCopied', this.i18nService.t('password')));
+    }
+
+    async submit() {
+        // Validation
+        if (this.newPassword == null || this.newPassword === '') {
+            this.platformUtilsService.showToast('error', this.i18nService.t('errorOccurred'),
+                this.i18nService.t('masterPassRequired'));
+            return false;
+        }
+
+        if (this.newPassword.length < 8) {
+            this.platformUtilsService.showToast('error', this.i18nService.t('errorOccurred'),
+                this.i18nService.t('masterPassLength'));
+            return false;
+        }
+
+        if (this.enforcedPolicyOptions != null &&
+            !this.policyService.evaluateMasterPassword(this.masterPasswordScore, this.newPassword,
+                this.enforcedPolicyOptions)) {
+            this.platformUtilsService.showToast('error', this.i18nService.t('errorOccurred'),
+                this.i18nService.t('masterPasswordPolicyRequirementsNotMet'));
+            return;
+        }
+
+        if (this.masterPasswordScore < 3) {
+            const result = await this.platformUtilsService.showDialog(this.i18nService.t('weakMasterPasswordDesc'),
+                this.i18nService.t('weakMasterPassword'), this.i18nService.t('yes'), this.i18nService.t('no'),
+                'warning');
+            if (!result) {
+                return false;
+            }
+        }
+
+        // Get user Information (kdf type, kdf iterations, resetPasswordKey, private key) and change password
+        try {
+            this.formPromise = this.apiService.getOrganizationUserResetPasswordDetails(this.organizationId, this.id)
+                .then(async response => {
+                    let kdfType = null;
```

Comment:
Small order of operations nit (but it's fine as-is, just perhaps in the future) when organizing logic, `const` handles memory better than `let` in the garbage collector although both handle context pretty well. Generally I would recommend testing the response first, if null throw an exception, THEN (outside of an `if/then` block) declare `const` and assign each of those variables. Makes things a little cleaner vs. assigning values which may never be used (declaring and assigning to `null` if we're only going to throw an exception anyway, etc.)

Reply:
Awesome bit of advice. Noted and updated.